src/main/java/duke/utils/TaskList.java

```diff
@@ -0,0 +1,161 @@
+package duke.utils;
+
+import duke.DukeException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * The `TaskList` class represents a list of tasks and provides methods for managing tasks within the list.
+ */
+public class TaskList {
+    private List<Task> toDoList;
```

Comment:
Perhaps naming it `taskList` is still better since `toDoList` might make people think that its a list of `ToDo`s instead of `Task`s?